server/s3_put_object_action.cc

```diff
@@ -623,6 +626,10 @@ void S3PutObjectAction::add_object_oid_to_probable_dead_oid_list() {
 
     // key = oldoid + "-" + newoid
     std::string old_oid_rec_key = old_oid_str + '-' + new_oid_str;
+
+    //prepending a char depending on the size of the object (size based bucketing of object)
+    S3CommonUtilities::size_based_bucketing_of_objects(old_oid_rec_key, object_metadata->get_content_length());
```

Comment:
This size_based_bucketing_of_objects should be only for object oid and not for key.